abcix/adapter/adapter_test.go

```diff
@@ -0,0 +1,59 @@
+package adapter
+
+import (
+	"testing"
+
+	"github.com/magiconair/properties/assert"
+	abci "github.com/tendermint/tendermint/abci/types"
+	abcix "github.com/tendermint/tendermint/abcix/types"
+)
+
+type mockAbciApp struct {
+	abci.BaseApplication
+}
+
+func (app *mockAbciApp) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
+	return abci.ResponseCheckTx{
+		Code: abci.CodeTypeOK,
+		Data: []byte("42"),
+	}
+}
+
+func (app *mockAbciApp) DeliverTx(req abci.RequestDeliverTx) abci.ResponseDeliverTx {
+	return abci.ResponseDeliverTx{
+		Data: req.Tx,
+	}
+}
+
+func (app *mockAbciApp) BeginBlock(req abci.RequestBeginBlock) abci.ResponseBeginBlock {
+	var events = make([]abci.Event, 1)
+	events[0] = abci.Event{Type: "begin"}
+	return abci.ResponseBeginBlock{Events: events}
+}
+
+func (app *mockAbciApp) EndBlock(req abci.RequestEndBlock) abci.ResponseEndBlock {
+	var events = make([]abci.Event, 1)
+	events[0] = abci.Event{Type: "end"}
+	return abci.ResponseEndBlock{Events: events}
+}
+
+func TestAdapt(t *testing.T) {
+	abciApp := &mockAbciApp{}
+	app := AdaptToABCIx(abciApp)
+
+	respCheckTx := app.CheckTx(abcix.RequestCheckTx{})
+	assert.Equal(t, respCheckTx.Code, abci.CodeTypeOK)
+	assert.Equal(t, respCheckTx.Data, []byte("42"))
+
+	var Txs = make([][]byte, 2)
+	Txs[0] = []byte("42")
+	Txs[1] = []byte("97")
+	respDeliverBlock := app.DeliverBlock(abcix.RequestDeliverBlock{Txs: Txs})
+	assert.Equal(t, respDeliverBlock.DeliverTxs[0].Data, Txs[0])
+	assert.Equal(t, respDeliverBlock.DeliverTxs[1].Data, Txs[1])
+
+	var events = make([]abcix.Event, 2)
+	events[0] = abcix.Event{Type: "begin"}
+	events[1] = abcix.Event{Type: "end"}
+	assert.Equal(t, respDeliverBlock.Events, events)
+}
```

Comment:
can you add a benchmark? something like the following

```
func (app *abciApp) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
  sleep(500ms)
}

func (app *abcixApp) CheckTx(req abcix.RequestCheckTx) abcix.ResponseCheckTx {
  sleep(500ms)
}
```

then adapt abci app to abcix and compare their time usage?

Reply:
Time usage for two `CheckTx`s: `BenchmarkAdaptedApp_CheckTx` calls `AdaptToABCIx()` and `BenchmarkAdaptedApp_CheckTx2` calls `abcix.CheckTx` directly.
![image](https://user-images.githubusercontent.com/60367163/90583578-60411780-e185-11ea-89c1-9c8c56d5bbe1.png)

Reply:
try sleeping 50ms?